go/packages/dev-mode/main_test.go

```diff
@@ -0,0 +1,279 @@
+}
+
+func extensionPlatformRuntimeDone(requestId string) {
+	payload := fmt.Sprintf(`{
+    "type": "platform.runtimeDone",
+		"record": {
+			"requestId": "%s",
+			"status": "success"
+		}
+	}`, requestId)
+	u.SubmitLogs([]byte(payload))
+}
+
+func postLogs(logs string) {
+	u.SubmitLogs([]byte(logs))
+}
+
+func TestSuccessfulLogForwarding(t *testing.T) {
+	wg := new(sync.WaitGroup)
+	wg.Add(1)
+	go func() {
+		ExternalExtension()
+		wg.Done()
+	}()
+
+	// First invocation
+	requestId := uuid.New().String()
+	extensionInvoke(requestId)
+	extensionReady()
+	extensionPlatformStart(requestId)
+
+	messages := []string{"1 invocation\n", "2 invocation \n"}
+	postLogs(fmt.Sprintf(`[
+		{
+			"type": "function",
+			"record": "%s"
+		},
+		{
+			"type": "function",
+			"record": "%s"
+		}
+	]`, messages[0], messages[1]))
+
+	validationData := getValidations()
+
+	// Validate we received the request id we thought we would
+	if validationData.RequestId != requestId {
+		t.Errorf("Expected requestId %s Received %s", requestId, validationData.RequestId)
+	}
+
+	// Validate we received the logs we thought we would
+	for _, payload := range validationData.Logs {
+		if payload.Name != functionName {
+			t.Errorf("Expected function name %s Received %s", functionName, payload.Name)
+			for index, message := range payload.Messages {
+				if message.Message != messages[index] {
+					t.Errorf("Expected log message %s Received %s", message.Message, messages[index])
+				}
+			}
+		}
+	}
+
+	extensionPlatformRuntimeDone(requestId)
+	// Reset  validations
+	resetValidations()
+
+	// Second invocation
+	requestId2 := uuid.New().String()
+	extensionInvoke(requestId2)
+	extensionPlatformStart(requestId2)
+
+	messages2 := []string{"3 invocation\n", "4 invocation \n"}
+	postLogs(fmt.Sprintf(`[
+		{
+			"type": "function",
+			"record": "%s"
+		},
+		{
+			"type": "function",
+			"record": "%s"
+		}
+	]`, messages2[0], messages2[1]))
+
+	validationData2 := getValidations()
+
+	// Validate we received the request id we thought we would
+	if validationData2.RequestId != requestId2 {
+		t.Errorf("Expected requestId %s Received %s", requestId2, validationData2.RequestId)
+	}
+
+	// Validate we received the logs we thought we would
+	for _, payload := range validationData2.Logs {
+		if payload.Name != functionName {
+			t.Errorf("Expected function name %s Received %s", functionName, payload.Name)
+			for index, message := range payload.Messages {
+				if message.Message != messages[index] {
+					t.Errorf("Expected log message %s Received %s", message.Message, messages2[index])
+				}
+			}
+		}
+	}
+
+	// End execution
+	extensionPlatformRuntimeDone(requestId2)
+	extensionShutdown(requestId2)
+	wg.Wait()
+}
+
+func TestRuntimeDoneBeforeInvoke(t *testing.T) {
+	wg := new(sync.WaitGroup)
+	wg.Add(1)
+	go func() {
+		ExternalExtension()
+		wg.Done()
+	}()
+
+	// First invocation
+	requestId := uuid.New().String()
+	extensionReady()
+	extensionPlatformRuntimeDone(requestId)
```

Comment:
This is not a natural layout. As `platform. runtimeDone` is never emitted before `platform.start`.

To reflect a real scenario, it should rather be tested with the first invocation going normally and following being reported in a way that we get `platform.start`, `platform.runtimeDone` and `invoke` (in that order).

Also it'll be good to have these where order is `platform.start`, `invoke` and `platform.runtimeDone`